src/parser/jobs/pld/modules/Faith.tsx

```diff
@@ -0,0 +1,72 @@
+import {Plural, Trans} from '@lingui/react'
+import {DataLink} from 'components/ui/DbLink'
+import {StatusKey} from 'data/STATUSES'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {EventHook} from 'parser/core/Dispatcher'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const TRACKED_STATUSES: StatusKey[] = [
+	'BLADE_OF_FAITH_READY',
+]
+
+export class Faith extends Analyser {
+	static override handle = 'faith'
+
+	@dependency private data!: Data
+	@dependency private suggestions!: Suggestions
+
+	private droppedFaiths: number = 0
+	private usedFaith: boolean = false
+
+	private faithHook?: EventHook<Events['action']>
+
+	override initialise() {
+		const trackedStatuses = TRACKED_STATUSES.map(key => this.data.statuses[key].id)
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		this.addEventHook(playerFilter.type('statusApply').status(oneOf(trackedStatuses)), this.onGain)
+		this.addEventHook(playerFilter.type('statusRemove').status(oneOf(trackedStatuses)), this.onDrop)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onGain() {
+		this.faithHook = this.addEventHook(
+			filter<Event>()
+				.source(this.parser.actor.id)
+				.type('action')
+				.action(this.data.actions.BLADE_OF_FAITH.id),
+			() => this.usedFaith = true,
+		)
+	}
+
+	private onDrop() {
+		if (!this.usedFaith) { this.droppedFaiths++ }
+		if (this.faithHook != null) {
+			this.removeEventHook(this.faithHook)
+			this.faithHook = undefined
+		}
+		this.usedFaith = false
+	}
```

Comment:
Standard approach, but since you only need to track if it's used or not you could likely just count status applies vs actions used as outright numbers directly in the hooks and just check if they're equal. WAR does basically the same thing but it used hooks due to tracking multiple statuses.

So, you could leave this as is (but it's overly complex for little benefit) and either track the raw numbers, or you could build this module on core Procs like I did to Primal Rend in #1558. NBD, just an observation.